engine/pkg/errctx/ctx.go

```diff
@@ -16,45 +16,57 @@ package errctx
 import (
 	"context"
 	"sync"
+	"time"
 )
 
 type errCtx struct {
-	context.Context
-	center *ErrCenter
+	ctx context.Context
```

Comment:
Why are we not embedding it directly?

Reply:
The [Uber Go Style Guide](https://github.com/uber-go/guide/blob/master/style.md#avoid-embedding-types-in-public-structs) doesn't suggest embedding types in public structs. But `errCtx` is not public struct, so it's okay to embed it directly.

Reply:
Done.